### PR TITLE
feat(session): add close and rejoin functions

### LINKED ISSUE #1815: feat: Implement  session_pause  and  session_resume
Add new proto messages ( UpdateParticipantStatePayload ,  RejoinRequestPayload ,  RejoinReplyPayload ) and implement  session_pause  and  session_resume.

See full design: #1807

### PR DESCRIPTION
**Pull request by micpapal**
# Description

This PR adds **close and rejoin** functionality to SLIM sessions, enabling participants in multicast sessions to gracefully go offline and come back online without leaving the session permanently.

### New Protocol Message
- Added `UpdateParticipantState` session message type and `UpdateParticipantStatePayload` to the protobuf schema, carrying participant name, new state (`ONLINE`/`OFFLINE`), and MLS epoch for validation.
- Added `ParticipantState` enum (`OnLine`, `OffLine`) to the protobuf definition.

### Session Controller API
- **`close()`** — notifies the group that this participant is going offline. Returns a `CompletionHandle` that resolves when all participants ACK. After completion, the session state is preserved for later rejoin. Only supported for multicast sessions.
- **`rejoin()`** — notifies the group that this participant is back online, including the current MLS epoch for validation. Participants verify epoch compatibility and ACK or NACK accordingly.
- Renamed the old `close()` (which permanently terminated the session) to **`leave()`** to clearly distinguish the two operations.

Close: https://github.com/agntcy/slim/issues/1815

## Type of Change

- [ ] Bugfix
- [x] New Feature
- [ ] Breaking Change
- [x] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](/agntcy/repo-template/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: crates/session/src/session_moderator.rs [hunk 1]
```diff
@@ -1193,6 +1267,188 @@ where
...
+        message: Message,
+        ack_tx: Option<oneshot::Sender<Result<(), SessionError>>>,
+    ) -> Result<SessionOutput, SessionError> {
+        // we can have only one update state/task at a time
+        if self.common.pending_status_update.is_some() || self.current_task.is_some() {
+            return Err(SessionError::StatusChangeInProgress);
+        }
+
+        // set the task to avoid to take other operation
+        self.current_task = Some(ModeratorTask::UpdateLocalStatus());
```
Comment (msardara) [inline crates/session/src/session_moderator.rs]: `current_task` is set to `UpdateLocalStatus()` before the fallible payload parse (`get_payload().ok_or(...)?`, `ParticipantState::try_from(...)?`) and before `send_control_message(...)`.

If any of those return `Err`, the task is left set with nothing to ever clear it, permanently blocking all future moderator tasks (Add/Remove/Close).

We should set `current_task` only after the payload is parsed and the send has succeeded, or clear it on the error paths.

File: crates/session/src/session_moderator.rs [hunk 2]
```diff
@@ -1193,6 +1267,188 @@ where
...
+                };
+
+                if let Some(epoch) = current_epoch
+                    && payload.epoch != epoch
+                {
+                    tracing::warn!(
+                        name = %participant_name,
+                        local_epoch = epoch,
+                        remote_epoch = payload.epoch,
+                        "epoch mismatch on rejoin, sending NACK",
```
Comment (msardara) [inline crates/session/src/session_moderator.rs:1356]: We should add a test with non-`None` epoch that results in epoch mismatch → NACK.

File: crates/session/src/session_controller.rs [hunk 3]
```diff
@@ -583,6 +585,69 @@ impl SessionController {
             .ok_or(SessionError::SessionAlreadyClosed)
     }
 
+    /// Close the session: notifies the group that this participant is going offline.
+    /// The returned CompletionHandle resolves when ACKs are collected.
```
Comment (msardara) [inline crates/session/src/session_controller.rs]: Nit: the doc says the handle "resolves when ACKs are collected", but it also resolves `Ok` on timeout (by design, per the `on_failure` OffLine branch). Worth stating so callers don't assume full-ACK guarantees.

File: crates/proto/proto/data-plane/v1/data_plane.proto [hunk 4]
```diff
@@ -303,6 +305,21 @@ message HeartbeatPayload {
   uint64 epoch = 1;
 }
 
+enum ParticipantState {
+  PARTICIPANT_STATE_ON_LINE = 0;
```
Comment (msardara) [inline crates/proto/proto/data-plane/v1/data_plane.proto]: I'd use `ONLINE` / `OFFLINE` (one word), what do you think?

**Review by msardara — CHANGES_REQUESTED** [review]

**Review by msardara — APPROVED** [review]
LGTM ✅
